resources/troubleshooting-guides/push_notification_troubleshooting.md

```diff
@@ -0,0 +1,171 @@
+### Set up a server key
+
+   1. To configure the app in Google Firebase, see [Firebase Console](https://console.firebase.google.com/) and sign in using your Google Developer credentials. 
+   2. Add a project and add your app under that project using the application ID that you used in the app.  
+   For more information, see [Add Firebase to your Android project](https://firebase.google.com/docs/android/setup#console).
+   3. Add the generated **“google-service.json”** to the project and set up the project and module level gradle files.  
+   For more information, see [Add Firebase to your Android project](https://firebase.google.com/docs/android/setup#console).
+   4. Verify that Firebase server key is configured in the Campaign instance by clicking **_Adobe Campaign -> Administrator -> Channels -> Mobile App (AEP SDK)_**.    .
+   5. Select the mobile app from the list and verify that the push channel settings status for Android is _Android key set up success_.  
+   
+   ![Server key configuration in Android](../../.gitbook/assets/android_server_key.png "Server key configuration in Android.")
+{% endtab %}
+
+{% tab title="iOS" %}
+
+### Set up the APNS server certificate
+
+   1. Add Push Notification entitlement to your app.  
+   For more information, see [Enable the Push Notifications Capability](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns?language=objc).
+   2. To configure the APNS certificate, register the app on your Apple developer account and generate the certificate.  
+   For more information, see [Establishing a Certificate-Based Connection to APNs](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_certificate-based_connection_to_apns?language=objc).
+   3. Verify that the APNS server certificate is configured in the Campaign instance by clicking **Adobe Campaign -> Administrator -> Channels -> Mobile App (AEP SDK)**.    
+   4. Select the mobile app from the list and verify that push channel settings status for iOS is _iOS certificate with filename *.pem is successfully uploaded_.  
+   
+ {% hint style="warning" %}
+ Verify that you are using production certificate for production environment and development certificate for development environment.
+ {% endhint %}
+   
+   ![Configured push certificate in Campaign](../../.gitbook/assets/campaign_ios_cert_configured.png "Configured push certificate in Campaign.")
+   
+{% endtab %}
+
+{% endtabs %}
+   
+For more information about configuration, see [Channel specific application configuration in Adobe Campaign.](https://helpx.adobe.com/campaign/kb/configuring-app-sdk.html#ChannelspecificapplicationconfigurationinAdobeCampaign).  
+  
+## Set up an Android or iOS app to receive push notifications  
+
+{% tabs %}
+{% tab title="Android" %}
+### Set up the Android app  
+
+   1. Create the Firebase Messaging Service and add it to the Android Manifest file.  
```

Comment:
These steps look like they are not flush left...please remove the spaces so that they appear flush left with the heading.